packages/flame/lib/src/effects/color_effect.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+
+import '../../components.dart';
+import 'component_effect.dart';
+import 'controllers/effect_controller.dart';
+
+/// Change the color of a component over time.
+///
+/// This effect applies incremental changes to the component's color, and
+/// requires that any other effect or update logic applied to the same component
+/// also used incremental updates.
+class ColorEffect extends ComponentEffect<HasPaint> {
+  final String? paintId;
+  final Color color;
+  late final Paint _original;
+  late final Tween<double> _tween;
+
+  ColorEffect(
+    this.color,
+    Offset offset,
+    EffectController controller, {
+    this.paintId,
+  })  : _tween = Tween(begin: offset.dx, end: offset.dy),
+        super(controller);
+
+  @override
+  Future<void> onMount() async {
+    super.onMount();
+
+    _original = target.getPaint(paintId);
+  }
+
+  @override
+  void apply(double progress) {
+    final currentColor = color.withOpacity(
+      _tween.transform(progress),
```

Comment:
I don't think you can take the progress here directly, you'll have to apply a delta, otherwise it won't mix if you run two `ColorEffect`s at the same time. Check how it is done here for example (but I don't think you'll have to care about the rounding errors):
https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/effects/opacity_effect.dart#L49-L52
This is maybe clearer:
https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/effects/move_effect.dart#L41-L45